deps/parser/parser.jl

```diff
@@ -0,0 +1,184 @@
+using JSON
+
+typestring = ""
+undotted_typestring = typestring[1:length(typestring)-1]
+
+type_translate_list = Dict(
+    "BigObject"          => typestring * "pm_perl_Object",
+    "OptionSet"          => typestring * "pm_perl_OptionSet",
+    "Matrix"             => typestring * "pm_Matrix{T} where T",
+    "Matrix<Rational>"   => typestring * "pm_Matrix{" * typestring * "pm_Rational}",
+    "Matrix<Integer>"    => typestring * "pm_Matrix{" * typestring * "pm_Integer}",
+    "Vector"             => typestring * "pm_Vector{T} where T",
+    "Vector<Rational>"   => typestring * "pm_Vector{" * typestring * "pm_Rational}",
+    "Vector<Integer>"    => typestring * "pm_Vector{" * typestring * "pm_Integer}",
+    "Integer"            => typestring * "pm_Integer",
+    "Rational"           => typestring * "pm_Rational",
+    "Set"                => typestring * "pm_Set",
+    "Anything"           => "Any"
+)
+
+## Dummy, will soon be generated
+filenames_list = Array{String,1}(
+    [ abspath( "tmp/polytope.json" ) ]
+)
+
+function julia_function_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_function( "$polymake_name$parameter_string", Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function julia_method_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_method( "$polymake_name$parameter_string", dispatch_obj, Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function create_argument_string_from_type( type_string::String, number::Int64 )
+    ## Special case for BigObject (we do not care 'bout the type here)
+    if startswith(type_string,"BigObject")
+        return "arg" * string(number) * "::" * type_translate_list["BigObject"]
+    end
+    ## We only translate the known cases
+    if !haskey(type_translate_list,type_string)
+        println("Unknown type: " * type_string)
+        return false
```

Comment:
here we could just throw an 
```julia
struct UnknownPolymakeType <: Exception
    msg::String
    UnknownPolymakeType(type_string) = new("Encountered an unknown polymake type: $type_string")
end
```

Reply:
This is not supposed to throw an exception, this is just supposed to warn me.

Reply:
then `@warn(...)`;
but You check for `falses` above, so this may as well throw with handling doing the warning